Atomic/discovery.py

```diff
@@ -21,17 +21,26 @@ def __init__(self, registry=None, repo=None, image=None, tag=None, digest=None,
         self.digest = digest
         self.orig_input = orig_input
         self._remote_inspect = None
-        self.fqdn = None
+        self._fqdn = None
 
         if self.debug:
             util.output_json(self.registries)
 
+    @property
+    def fqdn(self):
+        if not self._fqdn:
+            self.fqdn = self.assemble_fqdn(include_tag=True) if self.registry else self.find_image_on_registry()
+        return self._fqdn
+
+    @fqdn.setter
+    def fqdn(self, value):
+        self._fqdn = value
+
     def inspect(self):
         if self.registry:
-            self.fqdn = self.assemble_fqdn(include_tag=True)
             inspect_data = util.skopeo_inspect("docker://{}".format(self.fqdn), return_json=True)
         else:
-            self.fqdn = self.find_image_on_registry()
+            _ = self.fqdn
```

Comment:
theoretically yes because it sets the fqdn (the old code did this) and I didn't want to break a possible case I hadn't identified.  More of being safe than anything.